src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -129,17 +130,23 @@ INamedTypeSymbol typeSymbol
 			checker.CheckDeclaration( typeSymbol );
 		}
 
-		private static void AnalyzeGenericMethodTypeArguments(
+		private static void AnalyzeTypeArguments(
 			SyntaxNodeAnalysisContext ctx,
 			ImmutabilityContext immutabilityContext,
-			GenericNameSyntax syntax
+			SimpleNameSyntax syntax
 		) {
 			if( syntax.IsFromDocComment() ) {
 				// ignore things in doccomments such as crefs
 				return;
 			}
 
 			SymbolInfo info = ctx.SemanticModel.GetSymbolInfo( syntax, ctx.CancellationToken );
+
+			// Ignore anything that cannot have type arguments/parameters
+			if( info.Symbol is not INamedTypeSymbol && info.Symbol is not IMethodSymbol ) {

```

Comment:
I understand why this check is necessary but it's not totally obvious why... it's at least partially because `GetTypeParamsAndArgs` throws. It's a bit confusing to duplicate this check in two places.

Maybe we should change things a bit to use out vars and the `TryX` pattern instead of destructuring assignment: 

```cs
if ( !GetTypeParamsAndArgs( info.Symbol, out var typeParameters, out var typeArguments ) {
  // blah
  return;
}
```